### DISCUSSION TITLE
Suggestions on Adopting resources into helm release

### DISCUSSION THREAD

**Discussion by mikkergimenez** [grafana/tanka#663] [Q&A]
[This](https://github.com/helm/helm/pull/7649) change allows for the importing of existing resources into a helm release by adding managed-by labels.  Is there any thoughts on ways to run kubectl or add these kind of annotations using Tanka?  We're trying to start managing the aws-vpc-cni using Tanka, and it seems like the [only guidance so far ](https://github.com/aws/eks-charts/tree/master/stable/aws-vpc-cni#adopting-the-existing-aws-node-resources-in-an-eks-cluster) is to run these commands arbitrarily.  We might be able to put them into terraform, but we don't like the cross dependency between terraform and Tanka.  As it stands, the helm deploy will fail on the conflicting resources.  Also we might be able to run it all in terraform, but then we're running only one helm chart in terraform and all the rest in Tanka.

**Comment by Duologic** [comment]
Assuming from the linked PR, this seems to be about Helm running against a Kubernetes cluster with that intention of managing Kubernetes objects. Beware that the Helm integration in Tanka only reaches as far as `helm template`, Tanka does not run Helm against the Kubernetes cluster.

Having said that, you can add arbitrary annotations to your manifests with Jsonnet, here are a few helper functions that can iterate over all manifests: https://github.com/grafana/jsonnet-libs/tree/master/tanka-util#obj-k8s
